### DISCUSSION TITLE
Help on complex search queries

### DISCUSSION THREAD

**Discussion by nil0x42** [TriliumNext/Trilium#1651] [Q&A]
Hello world !

After trying many beta versions of trilium since 2019, with multiple changes in search operators since then, i am still frequently stuck trying to make an advanced search string for my `saved search` notes.

Here is the `saved search` i'm trying to make now:

`Notes with pending tasks`
```
note.type = text note.rawContent *=* '<input type="checkbox" disabled="disabled">' (#!cssClass OR #cssClass != todo)
```

The purpose of this search is to return all text notes that have at least one unchecked checkbox in their content.
The first part works, but i can't figure out how to make the last part of the query work as intended:
```
(#!cssClass OR #cssClass != todo)
```
What i want to do here, is to only return notes that do not have `#cssClass=todo` set.
But for some reason it returns completely different results.

Can you help me on this ?

**Comment by zadam** [comment]
Hi, your query looks correct.

It seems to be a bug in parsing `!=` operator. Fixed.

**Reply by nil0x42** [reply]
Thank you !

**Comment by nil0x42** [comment]
Okay, so while trying other complex search queries, i came into another one that seems to provide unexpected results:
```
#cssClass = todo OR (note.type = text AND note.rawContent *=* '<input type="checkbox" disabled="disabled">')
```

This search should return:
* all notes with `#cssClass=todo` label
* all text notes with a disabled checkbox

Instead it looks like the `#cssClass = todo` part is ignored

**Reply by zadam** [reply]
Hmm, tried the query and it seems to be working correctly. Is the note perhaps archived?

**Reply by nil0x42** [reply]
after trying v0.46.1-beta it is now working as expected

**Reply by zadam** [reply]
Perhaps it got incidentally fixed by some other change, not sure ...

**Reply by nil0x42** [reply]
Sometimes my saved searches including and entry with `#cssClass` start behaving weirdly, like ignoring the entry, showing results that souldn't be shown. Then some time pass, and next day it starts again.. Right now it's working as expected, but 2 days before, the axact same search returned weird results.

If it acts weird again i will send you the json debug of the search command. Maybe it helps.
